src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -1524,7 +1548,9 @@ void AlgorithmPaintToolBox::resetToolbox()
         updateButtons();
         m_magicWandButton->setChecked(false);
     }
-    deactivateCustomedCursor();
+
+    if (!isMeshData)
```

Comment:
Indeed since your last update with clear() we don't need it anymore but i did the first modification on an old commit where it was needed to switch between toolbox.